server/workflow/workflow_server.go

```diff
@@ -286,7 +286,7 @@ func (s *workflowServer) LintWorkflow(ctx context.Context, req *workflowpkg.Work
 
 	wftmplGetter := templateresolution.WrapWorkflowTemplateInterface(wfClient.ArgoprojV1alpha1().WorkflowTemplates(req.Namespace))
 
-	err := validate.ValidateWorkflow(wftmplGetter, req.Workflow, validate.ValidateOpts{})
+	err := validate.ValidateWorkflow(wftmplGetter, req.Workflow, validate.ValidateOpts{Lint: true})
```

Comment:
I believe this was missing during migration.

This is used here:
https://github.com/argoproj/argo/blob/master/workflow/validate/validate.go#L98